3_javascript/javascript1/week2/javascript.js

```diff
@@ -0,0 +1,102 @@
+// Formal fullname
+function getFullname(firstname, surname, useFormalName, isWoman) {
+    if (firstname === "" || surname === "") {
```

Comment:
Since `useFormalName` and `isWoman` are boolean you don't need the ===true comparison to check if they are true. You can simply call it like this 
```
if(useFormalName && isWoman)
{
...
}
```
Also the empty string `""`is a [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value in javascript.So you don't need the comparison there either because they are converted to boolean and will be false. 
So we can have something like this:

```javascript
// Formal fullname
function getFullname(firstname, surname, useFormalName, isWoman) {
    if (!firstname || !surname) {
        return `Please enter your full name`;
    }

    // Check if useFormalName and isWoman are both truthy
    if (useFormalName && isWoman) {
        return `Lady ${firstname} ${surname}`;
    }
    // Check if useFormalName is truthy and isWoman is falsy
    else if (useFormalName && !isWoman) {
        return `Lord ${firstname} ${surname}`;
    }
    else {
        return `${firstname} ${surname}`;
    }
}

Reply:
Very useful information. I wouldnt have known it.